src/encoding.rs

```diff
@@ -1,6 +1,45 @@
 use data_encoding::Encoding;
 use data_encoding_macro::new_encoding;
 
+// At a later point (probably when floating point arithmatic in const fn is stable) the below functions can be const
+
+#[cfg(feature = "alloc")]
+/// math comes from here https://github.com/bitcoin/bitcoin/blob/f1e2f2a85962c1664e4e55471061af0eaa798d40/src/base58.cpp
+pub(crate) fn gen_encoded_size(base: usize, input_byte_size: usize) -> usize {
+    (input_byte_size as f64 * (log10(256) / log10(base))) as usize + 1
```

Comment:
Would it be enough to calculate how many bits there are (`input_byte_size * 256`) and then dividing it by the number of bits used per character (the base) (`/ base`). Thus something along the lines of `(input_byte_size * 256) / base` and then round up to full bytes?

Please double check I'm not that much into base encoding.

Reply:
I will double check, but this is basically just the math copied from the commented link in the original base58 that base-x is built off of.

Reply:
I was mostly concerned about the custom `log()` implementation. But as we no longer need that I'm OK with an existing algorithm we know does the right thing.